randovania/gui/lib/area_picker.py

```diff
@@ -0,0 +1,262 @@
+        return isinstance(node, Node) and \
+            not self._isany(node, self._LOCATION_NODES + self._PICKUP_NODES + self._EVENT_NODES)
+
+    def filterAcceptsRow(self, source_row: int, source_parent: Union[QtCore.QModelIndex, QtCore.QPersistentModelIndex]) -> bool:
+        """
+        Determines if the current item should or should not be accepted into the model,
+        called automatically for each item.
+        """
+        source_index = self.sourceModel().index(source_row, 0, source_parent)
+        data = source_index.data(Qt.UserRole)
+        return any([filter(data) for filter in self._default_filters]) and \
+            self._is_in_valid_area(data)
+
+    def all_areas(self) -> Iterable[tuple[QtCore.QModelIndex, QtCore.QModelIndex]]:
+        root = self.index(0, 0)
+        i = 0
+        while self.hasIndex(i, 0, root):
+            world_index = self.index(i, 0, root)
+            j = 0
+            while self.hasIndex(j, 0, world_index):
+                area_index = self.index(j, 0, world_index)
+                yield world_index, area_index
+                j += 1
+            i += 1
+
+    def invalidateFilter(self):
+        """Invalidates the filter and updates it according to the current filter parameters."""
+        super().invalidateFilter()
+        self.parent().validate_root_indexes()
+
+    @staticmethod
+    def _isany(obj: object, types: list[type]):
+        """Checks if the object is any type in a list of types."""
+        return any ([isinstance(obj, t) for t in types])
+
+
+class AreaPickerDialog(QtWidgets.QDialog, Ui_AreaPickerDialog):
+    _model: AreaPickerModel
+    _world_list: WorldList
+    _empty_index: QtCore.QModelIndex
+    _pick_node: bool
+    worldList: QtWidgets.QListView
+    areaList: QtWidgets.QListView
+    nodeList: QtWidgets.QListView
+    world_combo_box: QtWidgets.QComboBox
+    area_combo_box: QtWidgets.QComboBox
+    node_combo_box: QtWidgets.QComboBox
+    confirm_buttons: QtWidgets.QDialogButtonBox
+    current_world: Optional[World] = None
+    current_area: Optional[Area] = None
+    current_node: Optional[Node] = None
+
+    def __init__(self, header_text: str, parent: QtWidgets.QWidget, game: RandovaniaGame, valid_areas: list[AreaIdentifier] = None, pick_node: bool = True, window_title: str = None):
+        super().__init__(parent)
+        self.setupUi(self)
+        world_list = default_database.game_description_for(game).world_list
+        self._pick_node = pick_node
+        self._model = AreaPickerModel(self, world_list, valid_areas)
+        self._setup_lists()
+        self.worldList.selectionModel().currentChanged.connect(self._update_areas)
+        self.searchLineEdit.textEdited.connect(self._update_search)
+        self.searchLineEdit.returnPressed.connect(self._confirm_selection)
+        self.confirmButtonBox.button(QtWidgets.QDialogButtonBox.Ok).setEnabled(False)
+        self.searchLineEdit.setFocus()
+        if window_title:
```

Comment:
if `window_title` is supposed to be optional, the check should be `window_title is not None`. (and updating the type hint to be `Optional[str]`.